### PR TITLE
Fixed some testing docs and consume in DeserializingConsumer

### PR DESCRIPTION
**Pull request by Egg3141592654**
This commit enables the consume method for the deserializing consumer. Encountering some issues running tests in WSL2, will create a VM and try to run tests fully there.

### CODE REVIEW

**Comment by Egg3141592654** [comment]
> It looks like @Egg3141592654 hasn't signed our **C**ontributor **L**icense **A**greement, yet.
> 
> > The purpose of a CLA is to ensure that the guardian of a project's outputs has the necessary ownership or grants of rights over all contributions to allow them to distribute under the chosen licence.
> > [Wikipedia](http://en.wikipedia.org/wiki/Contributor_License_Agreement)
> 
> You can read and sign our full Contributor License Agreement [here](http://clabot.confluent.io/cla).
> 
> Once you've signed reply with `[clabot:check]` to prove it.
> 
> Appreciation of efforts,
> 
> clabot

[clabot:check]

File: src/confluent_kafka/deserializing_consumer.py [hunk 1]
```diff
@@ -130,29 +130,83 @@ def poll(self, timeout=-1):
...
+        messages = super(DeserializingConsumer, self).consume(num_messages, timeout)
+
+        if messages is None:
+            return []
+
+        deserialized_messages = []
+
+        for message in messages:
+            if message.error() is not None:
+                raise ConsumeError(message.error(), kafka_message=message)
```
Comment (mhowlett) [inline src/confluent_kafka/deserializing_consumer.py:169]: this implementation effectively causes message without error to be thrown away, which is probably not ideal.

thanks for the PR, currently re-considering the need for batch consume.

File: tests/README.md [hunk 2]
```diff
@@ -18,7 +18,7 @@ A python3 env suitable for running tests:
 
     $ python3 -m venv venv_test
     $ source venv_test/bin/activate
-    $ pip install -r test/requirements.txt
+    $ pip install -r tests/requirements.txt
```
Comment (mhowlett) [inline tests/README.md:21]: 👍

File: .gitignore [hunk 3]
```diff
@@ -27,3 +27,4 @@ tests/docker/conf/tls/*
 .idea
 tmp-KafkaCluster
 .venv
+venv_test/*
```
Comment (mhowlett) [inline .gitignore:30]: 👍
but a newline would be nice.

Comment (edenhill) [reply .gitignore:30]: No need for `*`, the dir is sufficient.

File: src/confluent_kafka/deserializing_consumer.py (hunk 1 above)
Comment (Egg3141592654) [reply src/confluent_kafka/deserializing_consumer.py:169]: Good point, only did this to ensure that we kept parity with the singular API, but returning the messages with the error code is likely the better idea. I removed the exception throw.

**Comment by raphaelauv** [comment]
Nice I really hope this will be part of 1.8.0 :)

**Pull request closed without merging**